pkg/pool-weighted/contracts/InvariantGrowthProtocolFees.sol

```diff
@@ -51,14 +241,35 @@ abstract contract InvariantGrowthProtocolFees is BaseWeightedPool {
 
         uint256 preJoinExitInvariant = WeightedMath._calculateInvariant(normalizedWeights, preBalances);
 
-        uint256 toMint = WeightedMath._calcDueProtocolSwapFeeBptAmount(
-            totalSupply(),
+        uint256 supply = totalSupply();
+        uint256 swapFees = WeightedMath._calcDueProtocolSwapFeeBptAmount(
+            supply,
             _lastPostJoinExitInvariant,
             preJoinExitInvariant,
             protocolSwapFeePercentage
         );
 
-        _payProtocolFees(toMint);
+        uint256 yieldFees = 0;
+        if (_lastRateGrowthProduct > 0) {
+            uint256 rateGrowthProduct = WeightedMath._calculateWeightedProduct(normalizedWeights, _getRates());
+
+            // Only collect protocol fees when yield growth exceeds previous all-time high.
+            if (rateGrowthProduct > _lastRateGrowthProduct) {
+                yieldFees = WeightedMath._calcDueProtocolSwapFeeBptAmount(
+                    supply.add(swapFees),
+                    _lastRateGrowthProduct,
+                    rateGrowthProduct,
+                    // TODO: This fee pct should come from a different source.
+                    protocolSwapFeePercentage
+                );
+
+                _lastRateGrowthProduct = rateGrowthProduct;
+            }
+        } else {
+            _lastRateGrowthProduct = WeightedMath._calculateWeightedProduct(normalizedWeights, _getRates());
+        }
```

Comment:
It this because we need to measure growth relative to it? I'd add a comment here explaining that, and also mentioning that this will happen during pool initialization, so it's not like there's any unpaid fees.

Reply:
I'd also swap the order of the `if` and `else` bodies, for clarity.

Reply:
Correct. This variable is used to measure relative growth, so the check ensures that we don't take a fee before initialization. As you point out, this will be initialized during the INIT join, but the check is wasteful on every subsequent loop.

I marked a TODO in the constructor to consider initializing there instead, but it would be very messy given that we can't call `_getRates` and `_getNormalizedWeights` in the constructor.

Do you think it's worth exploring that option or should we just leave this here, add the comment, and flip the if/else order?